StackExchange.Redis/StackExchange/Redis/ExceptionFactory.cs

```diff
@@ -74,22 +74,46 @@ internal static Exception MultiSlot(bool includeDetail, Message message)
             if (includeDetail) AddDetail(ex, message, null, null);
             return ex;
         }
-
+
+        internal static string GetInnerMostExceptionMessage(Exception e)
+        {
+            if (e == null)
+            {
+                return "";
+            }
+            else
+            {
+                while (e.InnerException != null)
+                {
+                    e = e.InnerException;
+                }
+                return e.Message;
+            }
+        }
+        
         internal static Exception NoConnectionAvailable(bool includeDetail, RedisCommand command, Message message, ServerEndPoint server, ServerEndPoint[] serverSnapshot)
         {
-            string s = GetLabel(includeDetail, command, message);
+            string commandLabel = GetLabel(includeDetail, command, message);
 
             if (server != null)
             {
                 //if we already have the serverEndpoint for connection failure use that
                 //otherwise it would output state of all the endpoints
                 serverSnapshot = new ServerEndPoint[] { server };
             }
-            string exceptionmessage = "No connection is available to service this operation: " + s ;
-            var ex = new RedisConnectionException(ConnectionFailureType.UnableToResolvePhysicalConnection, exceptionmessage, GetServerSnapshotInnerExceptions(serverSnapshot));
+
+            var innerException = GetServerSnapshotInnerExceptions(serverSnapshot);
+
+            StringBuilder exceptionmessage = new StringBuilder("No connection is available to service this operation: ");
+            exceptionmessage.Append(commandLabel);
+            exceptionmessage.Append("; ");

```

Comment:
I'd get `GetInnerMostExceptionMessage(innerException)` in a variable, and if it's not `""`, then append. Right now we'd get a trailing `; ` in such cases.